### ISSUE TITLE
[Broken Website] Side-bar Add-on + twitter.com

### ISSUE THREAD

**Issue by takasismomo** [darkreader/darkreader#7130]
# Broken Website Report

## Website Address

[twitter.com](https://twitter.com)

## Steps To Reproduce

- install addon: [Side View](https://addons.mozilla.org/ja/firefox/addon/side-view/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)
- Go to twitter.com.
- Right click on the tab, Open in sidebar

## Expected Behavior

- Side bar has Same colors as on the tab.

## Actual Behavior

- Lost colors of icons.
- Lost user icons. 

## Screenshots

- Expected Behavior(Dark Reader is disabled on addon manager)
![ホーム _ Twitter — Mozilla Firefox 2021_10_19 16_56_04](https://user-images.githubusercontent.com/41175008/137867593-25487e20-024b-42d9-a21a-e856fb78400b.png)

- Actual Behavior(Dark Reader is enabled and turned off)
![ホーム _ Twitter — Mozilla Firefox 2021_10_19 16_55_10](https://user-images.githubusercontent.com/41175008/137867575-1498f9e4-12e8-483b-ad51-7abf0074b349.png)
## System Information

- OS: Windows10
- Browser: Firefox93.0
- Dark Reader Version: 4.9.39

**Comment by nivurek** [comment]
Try this:

https://github.com/darkreader/darkreader/issues/6882#issuecomment-939128800

**Comment by marlop352** [comment]
This seems to happen with any website I open in the sidebar, regardless of what addon I'm using to show a website on the sidebar, tested with:
https://addons.mozilla.org/en-US/firefox/addon/side-view/
https://addons.mozilla.org/en-US/firefox/addon/tweetdeck-for-sidebar/
https://addons.mozilla.org/en-US/firefox/addon/twitter-as-a-sidebar/
https://addons.mozilla.org/en-US/firefox/addon/open-link-in-sidebar/

Any tests I can do to help debug it?
Any about:config options that could be relevant?

### System Information
- OS: Windows11
- Browser: Firefox97.0
- Dark Reader Version: 4.9.45

( I'm using https://github.com/black7375/Firefox-UI-Fix/ )

**Comment by Gusted** [comment]
Only thing that is reproducible is that it's not getting modified, which isn't the problem that was originally reported:

![image](https://user-images.githubusercontent.com/25481501/195725998-6ae64658-c741-433b-834e-a0a9499ba494.png)

@marlop352 What kind of behavior are you having?

**Comment by marlop352** [comment]
The behavior I'm seeing is the same as on the original report, dark mode is always enabled for the sidebar unless the addon is disabled on the addon manager.

![dark reader](https://user-images.githubusercontent.com/846592/195726959-8313a0d5-f5a7-49f5-b1e4-989c0bedd98e.png)

If any information is needed to help diagnose this just say and I will provide it.
I can do any test needed on my profile and/or a clean profile as needed.

**Comment by Gusted** [comment]
@bershanskiy Could you confirm that the sender of these side-bars don't contain any information, such that the background script can send a message to it. I'm only seeing some vague data being present:
![image](https://user-images.githubusercontent.com/25481501/195729586-9513948f-7dc8-4b68-94ec-0feb1a314b88.png)

**Comment by bershanskiy** [comment]
The problem is caused by inability of Dark Reader's content script in sidebar to communicate with the background context. In particular:
 - Dark Reader does not respect "Site list" feature in sidebar because sidebar can not obtain the information about inclusion in the list
 - Dark Reader does not respect OFF toggle state because it is not communicated to sidebar
 - Dark Reader's style appears "broken" because it is actually just the "fallback style" used to prevent ["flash of unstyled (white) content"](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) and the real style is obtained from the background

> @bershanskiy Could you confirm that the sender of these side-bars don't contain any information, such that the background script can send a message to it. I'm only seeing some vague data being present:

Yes, this is exactly what I'm observing too. I actually started writing a patch for this yesterday which just boils down to using ports even for messages which do not require responses (like obtaining tab styles). Pretty ugly, to be honest and would prevent Dark Reader on Firefox from going "passive" once Firefox supports non-persistent pages properly. But I think this would work for now (until sidebars get their `sender.tab` or we change messaging again).

As a side note, Safari also has some oddities in messaging, for example `sender.url` is not supported. We might be able to get rid of substantial amount of messaging in #9212.

**Comment by Gusted** [comment]
> I actually started writing a patch for this yesterday which just boils down to using ports even for messages which do not require responses (like obtaining tab styles). Pretty ugly, to be honest and would prevent Dark Reader on Firefox from going "passive" once Firefox supports non-persistent pages properly.

Yeah that's indeed ugly, we don't want to interfere with browser behavior(especially when they are trying to resource limit). We could do another dirty trick and have a simple timeout(e.g. 1 second), if we don't hear back from the background script, clean up.

**Comment by bershanskiy** [comment]
I created a poll to access whether this bug is worth fixing, please consider casting your vote there: https://github.com/darkreader/darkreader/discussions/10105
